src/compiler/checker.ts

```diff
@@ -17219,6 +17303,28 @@ namespace ts {
             return true;
         }
 
+        function invocationError(node: Node, apparentType: Type, kind: SignatureKind) {
+            error(node, kind === SignatureKind.Call
+                ? Diagnostics.Cannot_invoke_an_expression_whose_type_lacks_a_call_signature_Type_0_has_no_compatible_call_signatures
+                : Diagnostics.Cannot_use_new_with_an_expression_whose_type_lacks_a_call_or_construct_signature
+            , typeToString(apparentType));
+            invocationErrorRecovery(apparentType, kind);
+        }
+
+        function invocationErrorRecovery(apparentType: Type, kind: SignatureKind) {
+            if (!apparentType.symbol) {
+                return;
+            }
+            const importNode = getSymbolLinks(apparentType.symbol).originatingImport;

```

Comment:
why can not we put the error on the call node itself? in the quickfix we can lookup the declaration of the symbol, and if found provide the fix there, if not, then it was aliased at some point, but i think that is fine to leave un-handled.

Reply:
Also can we do it in one error.. like, this cannot be called, because it comes from a namespace import..

@DanielRosenwasser thoughts?

Reply:
IMO, it makes sense to put the error where the code is likely wrong. In this case, the code is wrong to attempt to call or construct the thing, but that might be because it's the wrong thing; so we don't really know if the issue is because the import is wrong or because the usage is wrong, but we'd like to provide some hints as to how to fix it in either case.

Reply:
We talked about making the delayed error on the import on incorrect usage prior to starting to implement this feature because it seemed like a good compromise between always erroring on the import and having really bad errors on the usage.